gplog/gplog.go

```diff
@@ -120,31 +122,42 @@ func GetLogger() *GpLogger {
 // stdout and stderr are passed in to this function to enable output redirection in tests.
 func NewLogger(stdout io.Writer, stderr io.Writer, logFile io.Writer, logFileName string, verbosity int, program string) *GpLogger {
 	return &GpLogger{
-		logStdout:   log.New(stdout, "", 0),
-		logStderr:   log.New(stderr, "", 0),
-		logFile:     log.New(logFile, "", 0),
-		logFileName: logFileName,
-		verbosity:   verbosity,
-		header:      getHeader(program),
+		logStdout:     log.New(stdout, "", 0),
+		logStderr:     log.New(stderr, "", 0),
+		logFile:       log.New(logFile, "", 0),
+		logFileName:   logFileName,
+		verbosity:     verbosity,
+		header:        GetHeader(program),
+		logPrefixFunc: nil,
 	}
 }
 
-func getHeader(program string) string {
+func GetHeader(program string) string {
 	headerFormatStr := "%s:%s:%s:%06d-[%s]:-" // PROGRAMNAME:USERNAME:HOSTNAME:PID-[LOGLEVEL]:-
 	currentUser, _ := operating.System.CurrentUser()
 	user := currentUser.Username
 	host, _ := operating.System.Hostname()
 	pid := operating.System.Getpid()
 	header := fmt.Sprintf(headerFormatStr, program, user, host, pid, "%s")
 	return header
+}
 
+func SetLogPrefixFunc(logPrefixFunc func(string) string) {
+	logger.logPrefixFunc = logPrefixFunc
 }
 
-func GetLogPrefix(level string) string {
+func defaultLogPrefixFunc(level string) string {
```

Comment:
We might want to add a `ResetLogPrefix` function to set it back to `defaultLogPrefixFunc`.

Reply:
You can just set it to nil to restore the default logic.

Reply:
I was thinking in terms of the other change, to set `logPrefixFunc` to the default function instead of nil.  Since we have to leave that as nil for now, this is fine as-is.